File: c-tokens/source/comment_token.c
#include "./comment_token.h"

// Comment Tokens
const CommentToken BLOCK[3] = "/*";
const CommentToken LINE[3] = "//";
const CommentToken INNER_LINE[4] = "//!";
const CommentToken INNER_BLOCK[4] = "/*!";
const CommentToken OUTER_LINE[4] = "///";
const CommentToken OUTER_BLOCK[4] = "/**";
const CommentToken OCTOTHORPE_BLOCK[3] = "#*";
const CommentToken OCTOTHORPE_LINE[3] = "##";
const CommentToken OCTOTHROPE_INNER_LINE[3] = "#!";
const CommentToken OCTOTHORPE_INNER_BLOCK[4] = "#*!";
const CommentToken OCTOTHROPE_OUTER_LINE[4] = "###";
const CommentToken OCTOTHORPE_OUTER_BLOCK[4] = "#**";

// Comment Token Vector
const CommentToken comments_vector() {
    const CommentToken comments[12] = {
        BLOCK[3],
        LINE[3],
        INNER_LINE[4],
        INNER_BLOCK[4],
        OUTER_LINE[4],
        OUTER_BLOCK[4],
        OCTOTHORPE_BLOCK[3],
        OCTOTHORPE_LINE[3],
        OCTOTHROPE_INNER_LINE[3],
        OCTOTHORPE_INNER_BLOCK[4],
        OCTOTHROPE_OUTER_LINE[4],
        OCTOTHORPE_OUTER_BLOCK[4],
    };

    return *comments;
}

File: tokens/src/comment_token.rs
use std::{primitive::str, vec::Vec};

// Comment Tokens
pub const BLOCK: &'static str = "/*";
pub const LINE: &'static str = "//";
pub const INNER_LINE: &'static str = "//!";
pub const INNER_BLOCK: &'static str = "/*!";
pub const OUTER_LINE: &'static str = "///";
pub const OUTER_BLOCK: &'static str = "/**";
pub const OCTOTHORPE_BLOCK: &'static str = "#*";
pub const OCTOTHORPE_LINE: &'static str = "##";
pub const OCTOTHROPE_INNER_LINE: &'static str = "#!";
pub const OCTOTHORPE_INNER_BLOCK: &'static str = "#*!";
pub const OCTOTHROPE_OUTER_LINE: &'static str = "###";
pub const OCTOTHORPE_OUTER_BLOCK: &'static str = "#**";

// Comment Token Vector
pub fn comments_vector() -> Vec<&'static str> {
    let comments: Vec<&'static str> = Vec::from([
        BLOCK,
        LINE,
        INNER_LINE,
        INNER_BLOCK,
        OUTER_LINE,
        OUTER_BLOCK,
        OCTOTHORPE_BLOCK,
        OCTOTHORPE_LINE,
        OCTOTHROPE_INNER_LINE,
        OCTOTHORPE_INNER_BLOCK,
        OCTOTHROPE_OUTER_LINE,
        OCTOTHORPE_OUTER_BLOCK,
    ]);

    return comments;
}
